conan/tools/cmake/cmakedeps/templates/config.py

```diff
@@ -76,9 +75,9 @@ def template(self):
         {% if check_components_exist %}
         # Check that the specified components in the find_package(Foo COMPONENTS x y z) are there
         # This is the variable filled by CMake with the requested components in find_package
-        if({{ target_namespace }}_FIND_COMPONENTS)
-            foreach(_FIND_COMPONENT {{ '${'+target_namespace+'_FIND_COMPONENTS}' }})
-                if (TARGET {{ target_namespace }}::${_FIND_COMPONENT})
+        if({{ file_name }}_FIND_COMPONENTS)
+            foreach(_FIND_COMPONENT {{ '${'+file_name+'_FIND_COMPONENTS}' }})
```

Comment:
This was a bug in CMakeDeps

Reply:
what's the bug?

Reply:
That CMake generates a `<PackageName>_FIND_COMPONENTS` when `COMPONENTS` are specified in the `find_package`, but the `<PackageName>`!=`target_namespace`, it is `file_name`